docs/development/custom-apps.md

```diff
@@ -0,0 +1,134 @@
+    Platform: { History },
+} = Spicetify;
+
+// The main custom app render function. The component returned is what is rendered in Spotify.  
+function render() {
+    return react.createElement(Grid, { title: "My Custom App" });
+}
+
+// Our main component
+class Grid extends react.Component {
+    constructor(props) {
+        super(props);
+        Object.assign(this, props);
+        this.state = {
+            foo: "bar",
+            data: "etc"
+        };
+    }
+
+    render() {
+        return react.createElement("section", {
+            className: "contentSpacing",
+        },
+        react.createElement("div", {
+            className: "marketplace-header",
+        }, react.createElement("h1", null, this.props.title),
+        ),
+        ), react.createElement("div", {
+            id: "marketplace-grid",
+            className: "main-gridContainer-gridContainer",
+            "data-tab": CONFIG.activeTab,
+            style: {
+                "--minimumColumnWidth": "180px",
+            },
+        }, [...cardList]),
+        react.createElement("footer", {
+            style: {
+                margin: "auto",
+                textAlign: "center",
+            },
+        }, !this.state.endOfList && (this.state.rest ? react.createElement(LoadMoreIcon, { onClick: this.loadMore.bind(this) }) : react.createElement(LoadingIcon)),
+        ), react.createElement(TopBarContent, {
+            switchCallback: this.switchTo.bind(this),
+            links: CONFIG.tabs,
+            activeLink: CONFIG.activeTab,
+        }));
+    }
+}
+```
+
+## Common questions: 
+### My custom app isn't running when Spotify starts
+Your custom app will only run when it is clicked on the sidebar and its page loads. In order to run code on startup, you need to include a separate JS file as `subfiles_extension` in your [manifest](#manifest-file).
+### My subfile extension can't read my variables from my custom app
+Any subfile extensions are loaded separately from the main custom app, and do not have access to variables. You can use `localStorage` to save/load data between the two. 
+### How can I add a new sub page or path to my custom app?
+You can use `Spicetify.Platform.History.push(...)` to navigate to a new page. This can be a standard Spotify page, or a custom page for your app. You can include any data you need in the `state` key.
+```js
+Spicetify.Platform.History.push({
+    pathname: "/spicetify-marketplace/readme",
+    state: {
+        data: {
+            title: "My sub page title",
+            content: "My sub page content",
+        },
+    },
+});
+```
+
+In order to render a different page it is loaded, you can check the `pathname` of the current page. In the main render method in your `index.js`, you can check the `pathname`, and render a different page component for different paths. The main path for your custom app will be the name of the folder (what goes in your [`config-xpui.ini`](customization#configs)). 
```

Comment:
In order to render a different page ~~it is loaded~~, you can check the \`pathname\` of the current page **within** \`index.js\`'s main render method, ~~you can check the \`pathname\`,~~ and render a different page component for different paths. The main path for your custom app will be the name of the folder (**which is the same that needs to be used in the \[\`config-xpui.ini\`\](customization#configs) configuration file**).

Reply:
Updated.